### PR TITLE
[8.x][DROOLS-6679] Avoid NPE on PreProcess Interval evaluation

### PR DESCRIPTION
**Pull request by gitgabrio**
@danielezonca @mariofusco @jiripetrlik

See https://issues.redhat.com/browse/DROOLS-6679

### CODE REVIEW

File: kie-pmml-trusty/kie-pmml-commons/src/test/java/org/kie/pmml/commons/model/KiePMMLMiningFieldTest.java [hunk 1]
```diff
@@ -0,0 +1,95 @@
+package org.kie.pmml.commons.model;
```
Comment (danielezonca) [inline kie-pmml-trusty/kie-pmml-commons/src/test/java/org/kie/pmml/commons/model/KiePMMLMiningFieldTest.java:16]: Missing license variable

File: kie-pmml-trusty/kie-pmml-compiler/kie-pmml-compiler-commons/src/main/java/org/kie/pmml/compiler/commons/codegenfactories/KiePMMLMiningFieldFactory.java [hunk 2]
```diff
@@ -0,0 +1,234 @@
+/*
+ * Copyright 2020 Red Hat, Inc. and/or its affiliates.
```
Comment (danielezonca) [inline kie-pmml-trusty/kie-pmml-compiler/kie-pmml-compiler-commons/src/main/java/org/kie/pmml/compiler/commons/codegenfactories/KiePMMLMiningFieldFactory.java]:
```suggestion
 * Copyright 2021 Red Hat, Inc. and/or its affiliates.
```

File: kie-pmml-trusty/kie-pmml-models/kie-pmml-models-tree/kie-pmml-models-tree-tests/src/test/java/org/kie/pmml/models/tree/tests/TestTreeTest.java [hunk 3]
```diff
@@ -0,0 +1,58 @@
+package org.kie.pmml.models.tree.tests;
```
Comment (danielezonca) [inline kie-pmml-trusty/kie-pmml-models/kie-pmml-models-tree/kie-pmml-models-tree-tests/src/test/java/org/kie/pmml/models/tree/tests/TestTreeTest.java:16]: License header

File: kie-pmml-trusty/kie-pmml-models/kie-pmml-models-clustering/kie-pmml-models-clustering-model/src/main/resources/KiePMMLClusteringModelTemplate.tmpl [hunk 4]
```diff
@@ -21,6 +22,7 @@ public class KiePMMLClusteringModelTemplate extends KiePMMLClusteringModel {
         pmmlMODEL = null;
         transformationDictionary = null;
         localTransformations = null;
+         kiePMMLMiningFields = null;
```
Comment (danielezonca) [inline kie-pmml-trusty/kie-pmml-models/kie-pmml-models-clustering/kie-pmml-models-clustering-model/src/main/resources/KiePMMLClusteringModelTemplate.tmpl]:
```suggestion
        localTransformations = null;
        kiePMMLMiningFields = null;
```

**Review by danielezonca — APPROVED** [review]

**Review by mariofusco — APPROVED** [review]

File: kie-pmml-trusty/kie-pmml-compiler/kie-pmml-compiler-commons/src/main/java/org/kie/pmml/compiler/commons/codegenfactories/KiePMMLMiningFieldFactory.java (hunk 2 above)
Comment (gitgabrio) [reply kie-pmml-trusty/kie-pmml-compiler/kie-pmml-compiler-commons/src/main/java/org/kie/pmml/compiler/commons/codegenfactories/KiePMMLMiningFieldFactory.java]: @danielezonca 
done

File: kie-pmml-trusty/kie-pmml-commons/src/test/java/org/kie/pmml/commons/model/KiePMMLMiningFieldTest.java (hunk 1 above)
Comment (gitgabrio) [reply kie-pmml-trusty/kie-pmml-commons/src/test/java/org/kie/pmml/commons/model/KiePMMLMiningFieldTest.java:16]: @danielezonca
done

File: kie-pmml-trusty/kie-pmml-models/kie-pmml-models-tree/kie-pmml-models-tree-tests/src/test/java/org/kie/pmml/models/tree/tests/TestTreeTest.java (hunk 3 above)
Comment (gitgabrio) [reply kie-pmml-trusty/kie-pmml-models/kie-pmml-models-tree/kie-pmml-models-tree-tests/src/test/java/org/kie/pmml/models/tree/tests/TestTreeTest.java:16]: @danielezonca
done

File: kie-pmml-trusty/kie-pmml-models/kie-pmml-models-clustering/kie-pmml-models-clustering-model/src/main/resources/KiePMMLClusteringModelTemplate.tmpl (hunk 4 above)
Comment (gitgabrio) [reply kie-pmml-trusty/kie-pmml-models/kie-pmml-models-clustering/kie-pmml-models-clustering-model/src/main/resources/KiePMMLClusteringModelTemplate.tmpl]: @danielezonca 
Done

**Pull request merged**